RecoEgamma/EgammaPhotonProducers/src/ReducedEGProducer.cc

```diff
@@ -284,26 +291,28 @@ void ReducedEGProducer::produce(edm::Event& theEvent, const edm::EventSetup& the
   std::vector<std::vector<float> > gsfElectronPFClusterIsoVals(gsfElectronPFClusterIsoHandles.size());
   
   //loop over photons and fill maps
-  for (unsigned int ipho=0; ipho<photonHandle->size(); ++ipho) {
-    const reco::Photon &photon = (*photonHandle)[ipho];
+  for (const auto& photon : *photonHandle) {
```

Comment:
> it make more sense to me to set index=0 just above instead of 50 lines above

done.

> .. also, if you set "index = -1" outside of the loop, the first line in the loop can be "index++"
and there is no need to keep track of all possible places where the increment can be missed.

done.